common/src/java/org/apache/hadoop/hive/conf/HiveConf.java

```diff
@@ -850,6 +851,10 @@ public static enum ConfVars {
 
     HIVEIGNOREMAPJOINHINT("hive.ignore.mapjoin.hint", true, "Ignore the mapjoin hint"),
 
+    HIVE_CONF_LOCKED_LIST("hive.conf.locked.list", "", "Comma separated " +
+            "list of configuration options which are locked and can not be changed at runtime. Warning is logged and the " +
+            "change is ignored when user try to set these configs during runtime"),
```

Comment:
I guess this variable should be listed in the restricted list. Otherwise, users can bypass this option.
https://github.com/apache/hive/blob/539da19dc48740cb9c21cea0f8bff7ba0763e31b/common/src/java/org/apache/hadoop/hive/conf/HiveConf.java#L6996-L6999

Reply:
User cannot modify _hive.conf.locked.list_ because  it is not the part of whitelist (_hive.security.authorization.sqlstd.confwhitelist_) and whitelist cannot be modified at runtime. Moreover, _hive.conf.restricted.list_ checks are enforced only after the whitelist check. Here is an example:
```
0: jdbc:hive2://localhost:10000> set hive.conf.locked.list;
+-----------------------------------------------+
|                      set                      |
+-----------------------------------------------+
| hive.conf.locked.list=hive.auto.convert.join  |
+-----------------------------------------------+
1 row selected (0.022 seconds)
0: jdbc:hive2://localhost:10000> set hive.conf.locked.list=hive.execution.engine;
Error: Error while processing statement: Cannot modify hive.conf.locked.list at runtime. It is not in list of params that are allowed to be modified at runtime (state=42000,code=1)
0: jdbc:hive2://localhost:10000>
```

Reply:
There might be some security holes.

```
$ beeline
...
0: jdbc:hive2://hive-hiveserver2:10000/defaul> set hive.conf.locked.list;
+-------------------------------------+
|                 set                 |
+-------------------------------------+
| hive.conf.locked.list is undefined  |
+-------------------------------------+
1 row selected (0.179 seconds)
0: jdbc:hive2://hive-hiveserver2:10000/defaul> set hive.conf.locked.list=hive.execution.engine;
No rows affected (0.011 seconds)
0: jdbc:hive2://hive-hiveserver2:10000/defaul> set hive.conf.locked.list;
+----------------------------------------------+
|                     set                      |
+----------------------------------------------+
| hive.conf.locked.list=hive.execution.engine  |
+----------------------------------------------+
1 row selected (0.013 seconds)
```

Reply:
Looks like, that's because `hive.security.authorization.sqlstd.confwhitelist` is configured only by only a specific authorizer. I think it is better to put `hive.conf.locked.list` in `HiveConf#setupRestrictList` so that users using another authorizer can also use this feature safely.

https://github.com/apache/hive/blob/rel/release-4.0.0-beta-1/common/src/java/org/apache/hadoop/hive/conf/HiveConf.java#L6909-L6914

Reply:
Yes, I am also to reproduce this behaviour and have added hive.conf.locked.list to restricted list. Thanks for the thorough testing @okumin. Have update the PR.

Reply:
Looks good.

```
0: jdbc:hive2://hive-hiveserver2:10000/defaul> set hive.conf.locked.list;
+----------------------------------------------+
|                     set                      |
+----------------------------------------------+
| hive.conf.locked.list=hive.execution.engine  |
+----------------------------------------------+
1 row selected (0.012 seconds)
0: jdbc:hive2://hive-hiveserver2:10000/defaul> set hive.conf.locked.list=hive.auto.convert.join;
Error: Error while processing statement: Cannot modify hive.conf.locked.list at runtime. It is in the list of parameters that can't be modified at runtime or is prefixed by a restricted variable (state=42000,code=1)
```